client/ReviewModule.jsx

```diff
@@ -75,6 +76,64 @@ export default class ReviewModule extends React.Component {
     this.setState({ redVote: redArray, greyVote: greyArray });
   }
 
+  handleDropDown(e) {
+    const { reviews } = this.state;
+    const sortType = e.target.value;
+    let unsorted = reviews.slice();
+    const lowFirst = function compareLowest(a, b) {
+      return a.stars - b.stars;
+    };
+    const highFirst = function compareHighest(a, b) {
+      return b.stars - a.stars;
+    };
+    const earlyFirst = function compareEarliest(a, b) {
+      const yrA = parseInt(a.posted.slice(0, 4), 10);
+      const monthA = parseInt(a.posted.slice(5, 7), 10);
+      const dayA = parseInt(a.posted.slice(8, 10), 10);
+      const totalADays = yrA * 365 + monthA * 30 + dayA;
+      const yrB = parseInt(b.posted.slice(0, 4), 10);
+      const monthB = parseInt(a.posted.slice(5, 7), 10);
+      const dayB = parseInt(b.posted.slice(8, 10), 10);
+      const totalBDays = yrB * 365 + monthB * 30 + dayB;
+      if (totalADays < totalBDays) {
+        return -1;
+      }
+      if (totalADays > totalBDays) {
+        return 1;
+      }
+      return 0;
+    };
+    const lateFirst = function compareEarliest(a, b) {
+      const yrA = parseInt(a.posted.slice(0, 4), 10);
+      const monthA = parseInt(a.posted.slice(5, 7), 10);
+      const dayA = parseInt(a.posted.slice(8, 10), 10);
+      const totalADays = yrA * 365 + monthA * 30 + dayA;
+      const yrB = parseInt(b.posted.slice(0, 4), 10);
+      const monthB = parseInt(a.posted.slice(5, 7), 10);
+      const dayB = parseInt(b.posted.slice(8, 10), 10);
+      const totalBDays = yrB * 365 + monthB * 30 + dayB;
+      if (totalADays < totalBDays) {
+        return 1;
+      }
+      if (totalADays > totalBDays) {
+        return -1;
+      }
+      return 0;
+    };
+    if (sortType === 'Lowest') {
+      unsorted = unsorted.sort(lowFirst);
+    } else if (sortType === 'Highest') {
+      unsorted = unsorted.sort(highFirst);
+    } else if (sortType === 'Oldest') {
+      unsorted = unsorted.sort(earlyFirst);
+    } else if (sortType === 'Newest') {
+      unsorted = unsorted.sort(lateFirst);
+    } else {
+      this.setState({ reviews: unsorted });
```

Comment:
I don't think you need this line because you will call it later anyway. And it is kind of weird that you use the variable name unsorted even though they are sorted in the end. Not a big deal.

Reply:
Good catch.